topostats/tracing/disordered_tracing.py

```diff
@@ -384,6 +373,128 @@ def trace_image_disordered(  # pylint: disable=too-many-arguments,too-many-local
     return disordered_trace_crop_data, grainstats_additions_df, all_images, disordered_tracing_stats
 
 
+def compile_skan_stats(
+    skan_df: pd.DataFrame, skan_skeleton: skan.Skeleton, image: npt.NDArray, filename: str, grain_number: int
+) -> pd.DataFrame:
+    """
+    Obtain and add more stats to the resultant Skan dataframe.
+
+    Parameters
+    ----------
+    skan_df : pd.DataFrame
+        The statistics DataFrame produced by Skan's `summarize` function.
+    skan_skeleton : skan.Skeleton
+        The graphical representation of the skeleton produced by Skan.
+    image : npt.NDArray
+        The image the skeleton was produced from.
+    filename : str
+        Name of the file being processed.
+    grain_number : int
+        The number of the grain being processed.
+
+    Returns
+    -------
+    pd.DataFrame
+        A dataframe containing the filename, grain_number, branch-distance, branch-type, connected_segments,
+        mean-pixel-value, stdev-pixel-value, min-value, median-value, and mid-value.
+    """
+    skan_df["image"] = filename
+    skan_df["grain_number"] = grain_number
+    skan_df["connected_segments"] = skan_df.apply(find_connections, axis=1, skan_df=skan_df)
+    skan_df["min_value"] = skan_df.apply(lambda x: segment_heights(x, skan_skeleton, image).min(), axis=1)
+    skan_df["median_value"] = skan_df.apply(lambda x: np.median(segment_heights(x, skan_skeleton, image)), axis=1)
+    skan_df["mid_value"] = skan_df.apply(segment_middles, skan_skeleton=skan_skeleton, image=image, axis=1)
+
+    # remove unused skan columns
+    return skan_df[
+        [
+            "image",
+            "grain_number",
+            "branch-distance",
```

Comment:
Suggest using `_` in the variable names being added.

You can use `-` in Pandas variable names but it then prevents using the `.` (dot) notation to refer to an individual series of a dataframe.

I pointed this out on Skan and my pull request to convert it has been [merged](https://github.com/jni/skan/pull/215) but is awaiting a new release.

Reply:
cool I've renamed the columns but we'll need to remove that section when Skan updates :)